db/views/materialized_patient_summaries_v03.sql

```diff
@@ -0,0 +1,394 @@
+with latest_bp_passport as (
+    select distinct on (patient_id) *
+    from patient_business_identifiers
+    where identifier_type = 'simple_bp_passport' and deleted_at is null
+    order by patient_id, device_created_at desc
+), latest_phone_number as (
+    select distinct on (patient_id) *
+    from patient_phone_numbers
+    where deleted_at is null
+    order by patient_id, device_created_at desc
+), latest_medical_history as (
+    SELECT DISTINCT ON (patient_id) * FROM medical_histories WHERE deleted_at IS NULL
+),ranked_prescription_drugs as (
+    select
+        bp.id as bp_id,
+        array_agg(array[name, dosage] order by prescription_drugs.is_protocol_drug desc, prescription_drugs.name, prescription_drugs.device_created_at desc) as blood_pressure_drugs,
+        array_agg(name || '-' || dosage order by prescription_drugs.is_protocol_drug desc, prescription_drugs.name, prescription_drugs.device_created_at desc) as drug_strings
+    from blood_pressures bp
+        join prescription_drugs
+            on prescription_drugs.patient_id = bp.patient_id
+                and date(prescription_drugs.device_created_at) <= date(bp.recorded_at)
+                and (prescription_drugs.is_deleted is false or (prescription_drugs.is_deleted is true and date(prescription_drugs.device_updated_at) > date(bp.recorded_at)))
+    where bp.deleted_at is null and prescription_drugs.deleted_at is null
+    group by bp.id
+), blood_pressure_follow_up as (
+    select distinct on (bp.patient_id, date(bp.recorded_at)) bp.id bp_id, appointments.*
```

Comment:
Instead of using a lateral join to get the follow bp follow up, which would loop over the appointments table `O(no. of bps)` times, we can create a view of the latest follow up on the day of the BP and join using this table. The result set of the two queries is the same, and can be verified using 

```
with query1 as (
    select bp.id, a.id
    from blood_pressures bp
             left outer join (
        select distinct on (bp.patient_id, date(bp.recorded_at)) bp.id bp_id, appointments.*
        from blood_pressures bp
                 join appointments on appointments.patient_id = bp.patient_id and date(appointments.device_created_at) = date(bp.recorded_at)
        order by bp.patient_id, date(bp.recorded_at), appointments.device_created_at desc) a on a.bp_id  = bp.id
    where bp.deleted_at is null
), query2 as (
    select bp.id, a.id aid
    from blood_pressures bp
             left outer join lateral (
        select distinct on(patient_id) *
        from appointments
        where patient_id = bp.patient_id and date_trunc('day', device_created_at) = date_trunc('day', bp.recorded_at)
        order by patient_id, device_created_at desc
        ) a on a.patient_id = bp.patient_id
    where bp.deleted_at is null and a.deleted_at is null
)

(select * from query1 except  select * from query2) union all (select * from query2 except  select * from query1)
```